packages/react-native/React/Fabric/Mounting/ComponentViews/TextInput/RCTTextInputComponentView.mm

```diff
@@ -168,13 +168,20 @@ - (void)updateProps:(const Props::Shared &)props oldProps:(const Props::Shared &
     _backedTextInputView.autocapitalizationType =
         RCTUITextAutocapitalizationTypeFromAutocapitalizationType(newTextInputProps.traits.autocapitalizationType);
   }
+#endif // [macOS]
 
+#if !TARGET_OS_OSX // [macOS]
   if (newTextInputProps.traits.autoCorrect != oldTextInputProps.traits.autoCorrect) {
     _backedTextInputView.autocorrectionType =
         RCTUITextAutocorrectionTypeFromOptionalBool(newTextInputProps.traits.autoCorrect);
   }
-#endif // [macOS]
-
+#else // [macOS
+  if (newTextInputProps.traits.autoCorrect != oldTextInputProps.traits.autoCorrect && newTextInputProps.traits.autoCorrect.has_value()) {
+    _backedTextInputView.automaticSpellingCorrectionEnabled =
```

Comment:
The iOS path sets this property to a default value if `autoCorrect` does not have a value. Shouldn't macOS also keep the non-value?

Reply:
I think the difference is iOS is an enum while macOS is a simple bool. The type `newTextInputProps.traits.autoCorrect ` is an `std::optional<bool>` which is probably why the extra checking for a value and such.

Reply:
I think the correct behavior should be something like:
```objc
  if (newTextInputProps.traits.autoCorrect != oldTextInputProps.traits.autoCorrect) {
    _backedTextInputView.automaticSpellingCorrectionEnabled =
      newTextInputProps.traits.autoCorrect.has_value()
        ? newTextInputProps.traits.autoCorrect.value()
        : [[NSUserDefaults standardUserDefaults] boolForKey:@"NSAutomaticSpellingCorrectionEnabled"];
  }
```
But accessing `NSUserDefaults` today means we have to add a privacy manifest, so I'll let you decide whether it's worth it.

Reply:
No I would not want to do that. That's like.. making an sal query to set a prop. I'd rather keep as is. I vaguely remember this diff in paper too I think